tests/integration/long/test_loadbalancingpolicies.py

```diff
@@ -520,6 +520,89 @@ def test_token_aware_with_local_table(self):
 
         cluster.shutdown()
 
+    def test_token_aware_with_shuffle_rf2(self):
+        """
+        Test to validate the hosts are shuffled when the `shuffle_replicas` is truthy
```

Comment:
I see that this test asserts that the correct number of queries are made, but I don't see how it tests that they're shuffled. It's still a valuable test, but either this docstring is wrong or I don't understand the test.

I'm actually not sure how to test "replicas are shuffled". Maybe we should also add a regression test for PYTHON-643 -- test "if you make enough queries, some query will be issued to replicas in a different order than the first query was".